src/TextForEvent.js

```diff
@@ -17,11 +17,28 @@ import MatrixClientPeg from './MatrixClientPeg';
 import CallHandler from './CallHandler';
 import { _t } from './languageHandler';
 import * as Roles from './Roles';
+import dis from "./dispatcher";
+import React from 'react';
+
+function onUsernameClick(e) {
+    dis.dispatch({
+        action: 'insert_mention',
+        user_id: e.target.id,
+    });
+}
+
+function makeUsernameSpan(mxid, text) {
+    return <span className="mx_TextForEvent_username" dir="auto" onClick={onUsernameClick} id={mxid}>{ text }</span>;
```

Comment:
This isn't very friendly to keyboard users. You could make it an `a` and style it to make it look like normal text rather than making it text and styling it to `cursor: pointer`.

Reply:
I was basing it on sender Profile but that is a good point. Shall do